src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -283,6 +328,120 @@ public static void ValidateEntityNamesInConfig(Dictionary<string, Entity> entity
+                throw new DataApiBuilderException(
+                    message: $"Entity: {entityName} has a null rest {RestEntitySettings.PROPERTY_PATH}. Accepted value types are: string, boolean.",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                    );
+            }
+
+            if (restPathElement.ValueKind is not JsonValueKind.True && restPathElement.ValueKind is not JsonValueKind.False
+                && restPathElement.ValueKind is not JsonValueKind.String)
+            {
+                // The rest path can only be a string or a boolean value.
+                throw new DataApiBuilderException(
+                    message: $"Entity: {entityName} has rest {RestEntitySettings.PROPERTY_PATH} specified with incorrect data type. " +
+                    $"Accepted data types are: string, boolean.",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                    );
+            }
+
+            if (restPathElement.ValueKind is JsonValueKind.String)
+            {
+                string path = restPathElement.ToString().TrimStart('/').TrimStart(' ');
+                if (string.IsNullOrEmpty(path))
+                {
+                    // The rest 'path' cannot be empty.
+                    throw new DataApiBuilderException(
+                        message: $"Entity: {entityName} has an empty rest {RestEntitySettings.PROPERTY_PATH}.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                        );
+                }
+
+                return path;
+            }
+
+            return entityName;
+        }
+
+        /// <summary>
+        /// Helper method to validate that the Rest methods are correctly configured for the entity.
+        /// Rest methods can only be configured for stored procedures as an array of valid REST operations.
+        /// </summary>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="restMethodsElement">Rest methods element configured for the entity.</param>
+        /// <param name="entity">Entity object.</param>
+        /// <exception cref="DataApiBuilderException">Throws exception whenever a the rest methods are configured for a non-stored procedure entity or
+        /// contain an unexpected value.</exception>
+        private static void ValidateRestMethodsForEntity(string entityName, JsonElement restMethodsElement, Entity entity)
+        {
+            // This is needed to correctly populate the source type for the entity.
+            entity.TryPopulateSourceFields();
+            if (entity.ObjectType is not SourceType.StoredProcedure)
```

Comment:
Do we allow `"methods" : null` for non-stored procedure entities? should this verify first that restMethodsElement is not of JsonValueKind.Null?

Reply:
Yeah it shouldn't be allowed to specify this as null (if it is specified at all- we're fine if the methods property is omitted). The check will be done only for storedprocedure entities. For non-sp entities, we'll have the above exception.